presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/SpatialPartitioningStateSerializer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial;
+
+import com.facebook.presto.geospatial.Rectangle;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AccumulatorStateSerializer;
+import com.facebook.presto.spi.type.Type;
+import io.airlift.slice.BasicSliceInput;
+import io.airlift.slice.DynamicSliceOutput;
+
+import java.util.List;
+
+import static com.facebook.presto.geospatial.serde.GeometrySerde.readRectangle;
+import static com.facebook.presto.geospatial.serde.GeometrySerde.writeRectange;
+import static com.facebook.presto.plugin.geospatial.SpatialPartitioningType.SPATIAL_PARTITIONING;
+import static io.airlift.slice.SizeOf.SIZE_OF_DOUBLE;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+
+public class SpatialPartitioningStateSerializer
```

Comment:
For some reason we require serializer to be defined even if the aggregation function is not decomposable.

Create a stub, and add a TODO

```
public class SpatialPartitioningStateSerializer
        implements AccumulatorStateSerializer<SpatialPartitioningState>
{
    @Override
    public Type getSerializedType()
    {
        // TODO: make serializer optional in case of non decomposable aggregation
        return VARBINARY;
    }

    @Override
    public void serialize(SpatialPartitioningState state, BlockBuilder out)
    {
        throw new UnsupportedOperationException();
    }

    @Override
    public void deserialize(Block block, int index, SpatialPartitioningState state)
    {
        throw new UnsupportedOperationException();
    }
}
```